### PR TITLE
Write metadata-location instead of metadataLocation

### PR DESCRIPTION
**Pull request by Fokko**
While talking to the REST API I noticed that the API returns `metadataLocation`, instead of `metadata-location` as defined in the Open API Spec: https://github.com/apache/iceberg/blob/master/open-api/rest-catalog-open-api.yaml#L1486

### CODE REVIEW

File: core/src/main/java/org/apache/iceberg/rest/responses/LoadTableResponse.java [hunk 1]
```diff
@@ -89,8 +144,15 @@ public static class Builder {
     private Builder() {
     }
 
+      public Builder withMetadataLocation(String metadataLocation) {
+          this.metadataLocation = metadataLocation;
+          return this;
+      }
```
Comment (kbendick) [inline core/src/main/java/org/apache/iceberg/rest/responses/LoadTableResponse.java:150]: +1 to this. I find that sometimes, the `TableMetadata` that gets passed has `null` for `metadataLocation`, in which case we need to use the `TableMetadataBuilder` to add it which isn't very convenient.  👍

File: core/src/main/java/org/apache/iceberg/rest/responses/LoadTableResponse.java [hunk 2]
```diff
@@ -37,6 +42,56 @@
...
+    generator.writeStartObject();
+
+    generator.writeStringField(METADATA_LOCATION, loadTable.metadataLocation);
+
+    generator.writeFieldName(METADATA);
+    TableMetadataParser.toJson(loadTable.metadata, generator);
+
+    generator.writeObjectFieldStart(CONFIG);
+    for (Map.Entry<String, String> keyValue : loadTable.config.entrySet()) {
+      generator.writeStringField(keyValue.getKey(), keyValue.getValue());
```
Comment (kbendick) [inline core/src/main/java/org/apache/iceberg/rest/responses/LoadTableResponse.java:59]: Do we need a null check here? I believe this throws if `keyValue.getValue()` is null.

File: core/src/main/java/org/apache/iceberg/rest/responses/LoadTableResponse.java [hunk 3]
```diff
@@ -37,6 +42,56 @@
...
+
+  public static LoadTableResponse fromJson(JsonNode node) {
+    Preconditions.checkArgument(node.isObject(),
+            "Cannot parse table response from a non-object: %s", node);
+
+    String metadataLocation = null;
+    JsonNode  metadataLocationNode = node.get(METADATA_LOCATION);
+    if(metadataLocationNode != null) {
+      metadataLocation = metadataLocationNode.asText();
+    }
```
Comment (kbendick) [inline core/src/main/java/org/apache/iceberg/rest/responses/LoadTableResponse.java:74]: You can use `String metadataLocation = JsonUtil.getStringOrNull(METADATA_LOCATION, node)` for convenience here.

**Review by kbendick — COMMENTED** [review]
Technically this isn't presently needed, though the current strategy imposes the requirement that people use the `RESTObjectMapper` which has [kebab-case naming strategy.](https://github.com/apache/iceberg/blob/6a06cb3bd24c101d52e7987f55039b6b9692831d/core/src/main/java/org/apache/iceberg/rest/RESTObjectMapper.java#L43)

But this does make the requirement that people use the RESTObjectMapper or otherwise create one with the same characteristics, which is a limitation we should ideally remove as enforcing specific configuration on an ObjectMapper is hard to do in some frameworks without polluting things globally.

For reference, we can see that the read->write->read loop is tested in TestLoadTableResponse both [here in the round trip serialization comparison of the JSON string to itself](https://github.com/apache/iceberg/blob/6a06cb3bd24c101d52e7987f55039b6b9692831d/core/src/test/java/org/apache/iceberg/rest/responses/TestLoadTableResponse.java#L120). All of the request / response object tests also [check that the serialized JSON form only has the fields from the spec](https://github.com/apache/iceberg/blob/6a06cb3bd24c101d52e7987f55039b6b9692831d/core/src/test/java/org/apache/iceberg/rest/RequestResponseTestBase.java#L73-L96).


The fields from the spec come [from the test subclasses here (for LoadTableResponse)](https://github.com/apache/iceberg/blob/6a06cb3bd24c101d52e7987f55039b6b9692831d/core/src/test/java/org/apache/iceberg/rest/responses/TestLoadTableResponse.java#L71-L73). The test checks for both expected and unexpected fields, so that if something is added and not added to the test, we'll notice.

At present, we'd also need a serializer / deserializer for `CreateTableRequest` which is the other request / response object with multi-word keys at the moment that relies on the kebab-case naming strategy.

So overall I am +1 to this if we can get some simple tests for it and clean up the checkstyle issues 🙂

File: core/src/main/java/org/apache/iceberg/rest/responses/LoadTableResponse.java (hunk 2 above)
Comment (Fokko) [reply core/src/main/java/org/apache/iceberg/rest/responses/LoadTableResponse.java:59]: I copied this from the `TableMetadataParser` 🙈  Browsing through the code, I don't see any null checks.

File: core/src/main/java/org/apache/iceberg/rest/responses/LoadTableResponse.java (hunk 3 above)
Comment (Fokko) [reply core/src/main/java/org/apache/iceberg/rest/responses/LoadTableResponse.java:74]: Ooh, that's nice. Thanks!

File: core/src/main/java/org/apache/iceberg/rest/responses/LoadTableResponse.java (hunk 2 above)
Comment (rdblue) [reply core/src/main/java/org/apache/iceberg/rest/responses/LoadTableResponse.java:73]: I don't think there should be any null values.

**Comment by rdblue** [comment]
@Fokko, do you think we should move forward with this? As Kyle mentioned, if you use the included `RESTObjectMapper.mapper()` then the fields are sent and parsed correctly (as `metadata-location`). Did that solve the issue you ran into?

We were thinking that it wouldn't be necessary to add hand-coded parsers for each request/response class if we can ensure that the right mapper is always used. We do that in at least the client code.

**Comment by Fokko** [comment]
@rdblue I missed your comment here. I agree that we don't need the custom encoder, and I'll close this one. The bug was on our impl that has been fixed a while ago 👍🏻

**Pull request closed without merging**